src/compose.ts

```diff
@@ -366,7 +378,12 @@ function normalizeServiceVolumes(
 				break;
 		}
 	});
-	return [volumes, tmpfs];
+	appliedBindMountsByLabel.forEach(([label, appliedBindMounts]) => {
+		if (_.every(appliedBindMounts, (m) => mounts.indexOf(m) !== -1)) {
```

Comment:
The `mounts.indexOf(m) !== -1` would trigger if the mount is anywhere right?

So a mount like 

`/tmp:/var/run/docker.sock` will also cause the `balena-socket` label to be inserted.

Maybe something like this would be more accurate?
```suggestion
		if (_.every(appliedBindMounts, (m) => mounts.startsWith(m))) {
```

Reply:
Oh, but that sort of binding would get caught by the validation earlier, so maybe this is ok

Reply:
No, `mounts` is an array @pipex so this is doing a lookup, not a substring match.

Reply:
Got it, I missed that